app/wizards/a_level_steps/what_a_level_is_required.rb

```diff
@@ -2,23 +2,23 @@
 
 module ALevelSteps
   class WhatALevelIsRequired < DfE::Wizard::Step
-    # attr_accessor :answer
-    # validates :answer, presence: true
+    attr_accessor :subject, :other_subject, :minimum_grade_required
 
-    # def self.permitted_params
-    #   [:answer]
-    # end
+    Subject = Struct.new(:name, keyword_init: true)
 
-    # def previous_step
-    #   :first_step
-    # end
+    validates :subject, presence: true
+    validates :other_subject, presence: true, if: -> { subject == 'other_subject' }
+
+    def self.permitted_params
+      %i[subject other_subject minimum_grade_required]
+    end
+
+    def subjects_list
+      A_AND_AS_LEVEL_SUBJECTS.map { |name| Subject.new(name:) }
+    end
 
     # def next_step
-    #   if answer == 'yes'
-    #     :third_step
-    #   elsif answer == 'no'
-    #     :second_step
-    #   end
+    #  :exit
     # end
```

Comment:
There is not next step here because I am adding on the next card (still working on it). If everything is valid you will see a not implemented error. Again next card, next PR.